python/osrd_schemas/osrd_schemas/rolling_stock.py

```diff
@@ -169,6 +170,29 @@ class EtcsBrakeParams(BaseModel, extra="forbid"):
     t_be: float = Field(ge=0, description="T_be: safe brake build up time in s")
 
 
+class SimpleSignalingSystem(BaseModel, extra="forbid"):
+    signaling_system_type: Literal["BAL", "BAPR", "TVM300", "TVM430"] = Field(
+        alias="type"
+    )
+
+    model_config = ConfigDict(serialize_by_alias=True)
+
+
+class EtcsSignalingSystem(BaseModel, extra="forbid"):
+    signaling_system_type: Literal["ETCS_LEVEL2"] = Field(
+        alias="type", default="ETCS_LEVEL2"
+    )
+    brake_params: EtcsBrakeParams
+
+    model_config = ConfigDict(serialize_by_alias=True)
```

Comment:
After tests and discussion, on client's side, we want to **populate in code** using the **name** of the param (not alias) and we want to **serialize and deserialize** using **alias**.

So we end up on:
```suggestion
class SimpleSignalingSystem(BaseModel, extra="forbid"):
    signaling_system_type: Literal["BAL", "BAPR", "TVM300", "TVM430"] = Field(
        serialization_alias="type", validation_alias="type"
    )

    model_config = ConfigDict(serialize_by_alias=True, validate_by_name=True)


class EtcsSignalingSystem(BaseModel, extra="forbid"):
    signaling_system_type: Literal["ETCS_LEVEL2"] = Field(
        serialization_alias="type", validation_alias="type", default="ETCS_LEVEL2"
    )
    brake_params: EtcsBrakeParams

    model_config = ConfigDict(serialize_by_alias=True, validate_by_name=True)
```

Bumping pydantic is still necessary, let's keep it.

Reply:
It's sad that it has to be this complicated in python, while in rust it's just `serde(rename = ...)`

Reply:
✅ Tested locally and the client's pipeline agrees too.